crates/recursion/circuit-v2/src/merkle_tree.rs

```diff
@@ -67,7 +74,7 @@ impl<F: Field, HV: FieldHasher<F>> MerkleTree<F, HV> {
         (root, Self { height, digest_layers })
```

Comment:
Is there a reason why the root is not added to the merkle tree itself?

Reply:
I know you'll hate me for following Plonky3, but I think their distinction between Commitment and ProverData makes sense: the root is the commitment, and the ProverData should be what the prover needs to generate a proof. The root is not necessary for the prover to have after making the commitment.